application/src/main/java/uk/nhs/hee/tis/revalidation/messages/publisher/GmcSyncMessagePublisher.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2025 Crown Copyright (NHS England)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.revalidation.messages.publisher;
+
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GmcSyncMessagePublisher<T> extends RabbitMqMessagePublisher<T> {
+
+  @Value("${app.rabbit.reval.exchange}")
+  private String revalExchange;
+
+  @Value("${app.rabbit.reval.routingKey.gmcsync.requested.gmcclient}")
+  private String gmcSyncRoutingKey;
+
+  public GmcSyncMessagePublisher(String revalExchange, String gmcSyncRoutingKey,
+      RabbitTemplate rabbitTemplate) {
+    super(revalExchange, gmcSyncRoutingKey, rabbitTemplate);
+  }
+
+  @Override
+  public void publishToBroker(T message) {
```

Comment:
IDE complains if I don't explicitly call super, Sonar complains if I do, is this indicative of poor composition?

Reply:
May I ask what IDE complains?
It looks okay to remove this overrid method.